File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/AbstractGcpComputeBuilder.java
package com.sequenceiq.cloudbreak.cloud.gcp.compute;

import java.util.List;

import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
import com.sequenceiq.cloudbreak.cloud.gcp.AbstractGcpResourceBuilder;
import com.sequenceiq.cloudbreak.cloud.gcp.context.GcpContext;
import com.sequenceiq.cloudbreak.cloud.model.CloudInstance;
import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
import com.sequenceiq.cloudbreak.cloud.model.CloudResourceStatus;
import com.sequenceiq.cloudbreak.cloud.model.CloudStack;
import com.sequenceiq.cloudbreak.cloud.model.CloudVmInstanceStatus;
import com.sequenceiq.cloudbreak.cloud.template.ComputeResourceBuilder;

public abstract class AbstractGcpComputeBuilder extends AbstractGcpResourceBuilder implements ComputeResourceBuilder<GcpContext> {
    @Override
    public List<CloudResourceStatus> checkResources(GcpContext context, AuthenticatedContext auth, List<CloudResource> resources) {
        return checkResources(resourceType(), context, auth, resources);
    }

    @Override
    public List<CloudVmInstanceStatus> checkInstances(GcpContext context, AuthenticatedContext auth, List<CloudInstance> instances) {
        return null;
    }

    @Override
    public CloudVmInstanceStatus stop(GcpContext context, AuthenticatedContext auth, CloudInstance instance) {
        return null;
    }

    @Override
    public CloudVmInstanceStatus start(GcpContext context, AuthenticatedContext auth, CloudInstance instance) {
        return null;
    }

    @Override
    public CloudResource update(GcpContext context, CloudResource cloudResource, CloudInstance instance,
            AuthenticatedContext auth, CloudStack cloudStack) throws Exception {
        return null;
    }
}

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/AbstractGcpComputeBuilder.kt
package com.sequenceiq.cloudbreak.cloud.gcp.compute

import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext
import com.sequenceiq.cloudbreak.cloud.gcp.AbstractGcpResourceBuilder
import com.sequenceiq.cloudbreak.cloud.gcp.context.GcpContext
import com.sequenceiq.cloudbreak.cloud.model.CloudInstance
import com.sequenceiq.cloudbreak.cloud.model.CloudResource
import com.sequenceiq.cloudbreak.cloud.model.CloudResourceStatus
import com.sequenceiq.cloudbreak.cloud.model.CloudVmInstanceStatus
import com.sequenceiq.cloudbreak.cloud.template.ComputeResourceBuilder

abstract class AbstractGcpComputeBuilder : AbstractGcpResourceBuilder(), ComputeResourceBuilder<GcpContext> {

    override fun checkResources(context: GcpContext, auth: AuthenticatedContext, resources: List<CloudResource>): List<CloudResourceStatus> {
        return checkResources(resourceType(), context, auth, resources)
    }

    override fun checkInstances(context: GcpContext, auth: AuthenticatedContext, instances: List<CloudInstance>): List<CloudVmInstanceStatus>? {
        return null
    }

    override fun stop(context: GcpContext, auth: AuthenticatedContext, instance: CloudInstance): CloudVmInstanceStatus? {
        return null
    }

    override fun start(context: GcpContext, auth: AuthenticatedContext, instance: CloudInstance): CloudVmInstanceStatus? {
        return null
    }


}
